app/controllers/authentication_controller.rb

```diff
@@ -0,0 +1,15 @@
+###
+#  Controller for handling login/logout and redirecting with flash messages.
+###
+class AuthenticationController < ApplicationController
+  def login
+    session[:sunet] = request.env['REMOTE_USER'] || ENV['REMOTE_USER']
```

Comment:
This feels potentially problematic.  We typically don't store data like SUNet IDs in session data.

A more idiomatic approach might be to have a user class that models the attributes about a user (probably just SUNet ID right now).  That object can be instantiated in the controller context (e.g. `ApplicationController`) and be made available to all the views (typically under `current_user`).

This also follows the pattern of libraries like CanCan and devise, which would allow you to plug them in more easily of you so choose.